ls1intum/Artemis
# `Programming exercises`: Avoid nested zip for each submission repository

### Checklist
#### General


#### Server


#### Client
No changes in the client


#### Changes affecting Programming Exercises
- [x] I tested **all** changes and their related features with **all** corresponding user types on  ~~a test server~~ my local machine configured with **Gitlab** and **Jenkins**.


### Motivation and Context

If you download one or more student submission repositories to look at them with your IDE, 
you get a zip form Artemis, which contains several zip-files. One for each student.
The additional step to unzip is annoying and wastes time (see #7732).

(The creating of the sub-zip files also waste some computation power on the Artemis server.)

### Description
To not zip submission repositories, if they get zip together in the next step.
This applies to:
1. Download of submission to programming exercises at Scoring list
2. Archive a course with submissions to a programming exercises
3. Archive an exam with submissions to a programming exercises

(The exercise, test and solution repositories are still zipped again. So there is no breaking change in the download format for programming exercises, which can be imported again into Artemis.)


### Steps for Testing

Prerequisites:
- 1 Instructor
- 2 Students
- 1 Course

1. Log in to Artemis with the Instructor account
2. Navigate to Course Administration of your Course
3. Create a new programming exercise (recommendation: :heavy_check_mark:  Allow Online Editor)
4. Make a submission with both Student account
5.  Go to the Scores page of the exercise
6. Click `Export` - `Download Repos`
7. Download the repositories of all students (or maybe also try only one student)
8. Check that the downloaded zip contains **directories** for each student and not **zip files**
9. Navigate to Course Administration of your Course
10. Set the end date of your course to now
11.  Click on `Archive Course`
12. Confirm the message and wait for the progress to finish
13. Click on `Download Archive`
14. In the zip file you can find a directory for your exercise. Check that it contains **directories** for each student and not **zip files** (zip files for exercise, test and solution are by design.)

#### Exam Mode Testing



Prerequisites:
- 1 Instructor
- 2 Students

1. Log in to Artemis with the Instructor account
2. Create an exam for one exercise in it
3. Create an exercise group and a programming exercise in it with 1 maximal point (recommendation: :heavy_check_mark:  Allow Online Editor)
4. Add both student account to the exam and generate their exams
5. Participant in the exam with both student accounts including a submission to the exercise
6. Set all ends of the exam to now
7. Archive the exam
8. Click on `Archive Exam`
9. Confirm the message and wait for the progress to finish
10. Click on `Download Archive`
11.  In the zip file you can find a directory for your exercise. Check that it contains **directories** for each student and not **zip files** (zip files for exercise, test and solution are by design.)

### Testserver States
> [!NOTE]
> These badges show the state of the test servers.
> Green = Currently available, Red = Currently locked

![](https://byob.yarr.is/ls1intum/Artemis/artemis-test1)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test2)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test3)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test4)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test5)
![](https://byob.yarr.is/ls1intum/Artemis/artemis-test6)

### Review Progress






#### Performance Review
#### Code Review
#### Manual Tests
#### Exam Mode Test

### Test Coverage




| Class/File                                                         | Line Coverage | Confirmation (assert/expect) |
|------------------------------------------------------------|----------------------:|------------------------------------------:|
| GitService.java                                               | 85.10%              | ✅                                                |
| AthenaRepositoryExportService.java        | 94.74%              | ✅                                                |
| ProgrammingExerciseExportService.java | 69.47%              | ✅                                                |
